Add gren column

blaix opened the pull request:
Very similar to elm column. Biggest differences are:

* No tuples
* No lists
* No `::`

blaix committed:
Add gren column

Very similar to elm column. Biggest differences are:

* No tuples
* No lists
* No `::`

blaix commented on transformations.html:
```diff
@@ -591,6 +761,24 @@ <h2>Pattern matching</h2>
+case xs of
+    []     -> Nothing
+    [a, b] -> Just { first = a, rest = [b] }
+    _ -> Array.popFirst xs
```
This one is a little awkward, since you wouldn't do this in gren. There is no `::` and `popFirst` is already doing what the entire case expression is doing. But wrote it like this to match the syntax examples.

supermario replied:
@blaix I think it's fine to write it like it should be in Gren! The goal for this reference is to give an idiomatic sense too, not just be a pure conversion at the expense of sense.

And if literal conversion is fine but an alternative is preferred for some reason, I think a second example with a note is great 👍 

Leave it to you to decide.

blaix replied:
Changed to a more idiomatic example with a comment about lack of `::`

blaix committed:
Gren doesn't have List

Missed this in the first pass.

blaix committed:
Update gren pattern matching to be more idiomatic.

Show a realistic example instead of re-implementing the behavior of
Array.popFirst just to match the shape of the other examples.

supermario commented:
Nice one, thanks @blaix !
